src/parcsr_mv/par_csr_matop.c

```diff
@@ -4161,7 +4161,6 @@ hypre_ParTMatmul( hypre_ParCSRMatrix  *A,
    if ( hypre_GetExecPolicy2(memory_location_A, memory_location_B) == HYPRE_EXEC_DEVICE )
    {
       hypre_CSRMatrixMoveDiagFirstDevice(hypre_ParCSRMatrixDiag(C));
-      hypre_SyncComputeStream();
```

Comment:
why removing this line?

Reply:
If you look in `hypre_CSRMatrixMoveDiagFirstDevice()`, there is a sync at the end of that function already.

Reply:
Ah, OK. Good point. Thanks!